src/statistiek_hub/modeladmins/measure_admin.py

```diff
@@ -86,22 +82,32 @@ class MeasureAdmin(ImportExportFormatsMixin, admin.ModelAdmin):
         ),
     )
 
+    inlines = [FilterInline]
+
     def get_readonly_fields(self, request, obj=None):
         if request.user.is_superuser:
             return []
         return self.readonly_fields
 
-    def get_queryset(self, request):
-        qs = super().get_queryset(request)
-        if request.user.is_superuser:
-            return qs
-        return qs.filter(owner=request.user)
+    def _get_user_groups(self, request):
+        # Collect user groups once
+        if not hasattr(request, '_cached_user_groups'):
+            request._cached_user_groups = request.user.groups.all()
+        return request._cached_user_groups
 
-    # measure field 'owner' get's filled automaticaly
-    def save_model(self, request, obj, form, change):
-        if not obj.pk:
-            # Only set owner during the first save.
-            obj.owner = request.user
-        super().save_model(request, obj, form, change)
+    def has_change_permission(self, request, obj=None):
+        if obj is not None:
+            user_groups = self._get_user_groups(request)
+            if obj.theme.group in user_groups:
+                return True
+            return False
```

Comment:
It looks good but you know me ;)
```suggestion
            return obj.theme.group in self._get_user_groups(request)
```

Less code to parse makes it easier to read for me. What do you think?